uportal-war/src/main/java/org/jasig/portal/portlets/portletadmin/PortletAdministrationHelper.java

```diff
@@ -158,87 +153,48 @@
-		    form = new PortletDefinitionForm(def);
-		    form.setId(def.getPortletDefinitionId().getStringId());
-		}
-		else {
-		    form = new PortletDefinitionForm();
-		}
-
-		// if this is a pre-existing portlet, set the category and permissions
+        // create the new form
+        final PortletDefinitionForm form;
         if (def != null) {
+            // if this is a pre-existing portlet, set the category and permissions
+            form = new PortletDefinitionForm(def);
+            form.setId(def.getPortletDefinitionId().getStringId());
 
-        	// create a JsonEntityBean for each current category and add it
-        	// to our form bean's category list
-        	Set<PortletCategory> categories = portletCategoryRegistry.getParentCategories(def);
-        	for (PortletCategory cat : categories) {
-        		form.addCategory(new JsonEntityBean(cat));
-        	}
-
-			try {
-                IPermissionManager pm = AuthorizationService.instance().newPermissionManager(IPermission.PORTAL_SUBSCRIBE);
-                IAuthorizationPrincipal[] prins =
-                        pm.getAuthorizedPrincipals(IPermission.PORTLET_SUBSCRIBER_ACTIVITY,
-                        PermissionHelper.permissionTargetIdForPortletDefinition(def));
-                for (int mp = 0; mp < prins.length; mp++) {
-                	JsonEntityBean bean;
-
-                	// first assume this is a group
-                	IEntityGroup group = GroupService.findGroup(prins[mp].getKey());
-                	if (group != null) {
-                    	bean = new JsonEntityBean(group, EntityEnum.GROUP);
-                	}
-
-                	// if a matching group can't be found, try to find a matching
-                	// non-group entity
-                	else {
-                    	IGroupMember member = AuthorizationService.instance().getGroupMember(prins[mp]);
-                    	bean = new JsonEntityBean(member, EntityEnum.PERSON);
-                    	String name = groupListHelper.lookupEntityName(bean);
-                    	bean.setName(name);
-                	}
-
-                    form.addGroup(bean);
-                }
-			} catch (GroupsException e) {
-				e.printStackTrace();
-			}
-		}
+            // create a JsonEntityBean for each current category and add it
+            // to our form bean's category list
+            Set<PortletCategory> categories = portletCategoryRegistry.getParentCategories(def);
+            for (PortletCategory cat : categories) {
+                form.addCategory(new JsonEntityBean(cat));
+            }
 
-        // otherwise, if this is a new portlet, pre-populate the categories
-        // and groups with some reasonable defaults
+            try {
+                getGroupPermissions(def, form, IPermission.PORTLET_SUBSCRIBER_ACTIVITY);
+                getGroupPermissions(def, form, IPermission.PORTLET_BROWSE_ACTIVITY);
+            } catch (GroupsException e) {
+                e.printStackTrace();
```

Comment:
There are no checked exceptions thrown from getGroupPermissions() -- removing the try-catch block.